hadoop-project/pom.xml

```diff
@@ -2737,6 +2737,46 @@
         </dependencies>
       </dependencyManagement>
     </profile>
+    <profile>
+      <id>jdk17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <properties>
+        <maven.compiler.release>17</maven.compiler.release>
+        <argLine>
+          --add-opens java.base/jdk.internal.ref=ALL-UNNAMED
```

Comment:
where does the list come from?

Reply:
Thank you for your question! This list comes from the additional compilation items required by our internal Hadoop version to support JDK 17. Currently, the list might contain some redundancy as the community is continuously working on JDK 17 improvements, and our internal version has not merged all the latest PRs from the community.

Reply:
How does this interact with existing setting of `argLine`?

https://github.com/apache/hadoop/blob/3f637efaa2ce02759df213b9f1aa747614f4167a/hadoop-project/pom.xml#L2459-L2463

https://github.com/apache/hadoop/blob/3f637efaa2ce02759df213b9f1aa747614f4167a/hadoop-project/pom.xml#L170-L177

Does it simply override the value coming from `maven-surefire-plugin.argLine`?  If so, can we tweak that property instead, to avoid confusion?  With `-XX:+IgnoreUnrecognizedVMOptions` we can do that regardless of Java version (i.e. outside of this profile).

Reply:
@slfan1989 @adoroszlai I made a more accurate JPMS option list at https://github.com/apache/hadoop/pull/7114 by running all UTs with JDK 17, and added them to global `maven-surefire-plugin.argLine` with the option `-XX:+IgnoreUnrecognizedVMOptions`

Reply:
@pan3793 Thank you for your contribution! I plan to directly incorporate `${maven-surefire-plugin.argLine}` into the `<argLine>` after it is merged in #7114.

Reply:
```
<profile>
      <id>jdk9</id>
      <activation>
        <jdk>[9,)</jdk>
      </activation>
      <properties>
        <maven.compiler.release>17</maven.compiler.release>
        <argLine>
          $${maven-surefire-plugin.argLine}
        </argLine>
      </properties>
      <build>
        <pluginManagement>
          <plugins>
            <plugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-javadoc-plugin</artifactId>
              <configuration>
                <source>${javac.version}</source>
                <additionalOptions>--ignore-source-errors</additionalOptions>
                <failOnError>false</failOnError>
              </configuration>
            </plugin>
          </plugins>
        </pluginManagement>
      </build>
    </profile>
```

Reply:
My suggestion is just

```patch
+   <profile>
+      <id>java9</id>
+      <activation>
+        <jdk>[9,)</jdk>
+      </activation>
+      <properties>
+        <maven.compiler.release>${javac.version}</maven.compiler.release>
+      </properties>
+    </profile>
```

and it's encouraged to

> use version numbers without the "1." prefix (supported since javac 5)

https://github.com/apache/maven-apache-parent/commit/ded34a863110021191af2db5b772b665a2789394#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R101

```patch
-   <javac.version>1.8</javac.version>
+   <javac.version>8</javac.version>
```

Reply:
I readed the document provided by Pan, and I find it reasonable.

```
-   <javac.version>1.8</javac.version>
+   <javac.version>8</javac.version>
```

cc: @steveloughran